src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -749,7 +749,21 @@ private Boolean IsOpenable(Boolean needToUncompress, Boolean needToLoadIntoMemor
                     if (CompressionMethod != CompressionMethodValues.Stored &&
                         CompressionMethod != CompressionMethodValues.Deflate)
                     {
-                        message = SR.UnsupportedCompression;
+                        switch (CompressionMethod)
+                        {
+                            case CompressionMethodValues.Deflate64:
+                                message = SR.UnsupportedCompression + " Deflate64";
+                                break;
+                            case CompressionMethodValues.BZip2:
+                                message = SR.UnsupportedCompression + " Bzip2";
+                                break;
+                            case CompressionMethodValues.LZMA:
+                                message = SR.UnsupportedCompression + " LZMA";
+                                break;
+                            default:
+                                message = SR.UnsupportedCompression;
```

Comment:
Sure, we could do that instead.

There are some values that don't have an associated compression 'name' or are seldom used, so for those I figured we'd keep with the current message. To do that, we'd want to add another SR message that has the parameter and leave the current one intact.

Or we could just print the value itself and not worry about the string name i.e. print "Unsupported compression value 9" instead of "Unsupported compression method Deflate64".

I'll push an update with the parameter.